src/diagnostics/default_diagnostics.jl

```diff
@@ -69,130 +79,54 @@ function average_pre_output_hook!(accum, counter)
-
-"""
-    daily_averages(short_names...; output_writer)
-
-
-Return a list of `ScheduledDiagnostics` that compute the daily average for the given variables.
-"""
-# An average is just a sum with a normalization before output
-daily_averages(short_names...; output_writer) = common_diagnostics(
-    24 * 60 * 60,
-    (+),
-    output_writer,
-    short_names...;
-    pre_output_hook! = average_pre_output_hook!,
-)
-"""
-    daily_average(short_names; output_writer)
-
-
-Return a `ScheduledDiagnostics` that compute the daily average for the given variable.
-"""
-# An average is just a sum with a normalization before output
-daily_average(short_names; output_writer) =
-    daily_averages(short_names; output_writer)[1]
-"""
-    hourly_maxs(short_names...; output_writer)
-
-
-Return a list of `ScheduledDiagnostics` that compute the hourly max for the given variables.
-"""
-hourly_maxs(short_names...; output_writer) =
-    common_diagnostics(60 * 60, max, output_writer, short_names...)
-
-"""
-    hourly_max(short_names; output_writer)
-
-
-Return a `ScheduledDiagnostics` that computes the hourly max for the given variable.
-"""
-hourly_max(short_names...; output_writer) =
-    hourly_maxs(short_names; output_writer)[1]
-
-"""
-    hourly_mins(short_names...; output_writer)
-
+include("standard_diagnostic_frequencies.jl")
 
-Return a list of `ScheduledDiagnostics` that compute the hourly min for the given variables.
-"""
-hourly_mins(short_names...; output_writer) =
-    common_diagnostics(60 * 60, min, output_writer, short_names...)
 """
-    hourly_mins(short_names...; output_writer)
+    frequency_averages(t_end::Real)
 
+Return the correct averaging function depending on the total simulation time.
 
-Return a `ScheduledDiagnostics` that computes the hourly min for the given variable.
-"""
-hourly_min(short_names; output_writer) =
-    hourly_mins(short_names; output_writer)[1]
+If `t_end < 1 day` take hourly means,
+if `t_end < 120 days` take means over ten days,
+if `t_end < 1 year` take daily means,
+If `t_end >= 1 year` take monthly means.
 
-# An average is just a sum with a normalization before output
+One month is defined as 30 days.
 """
-    hourly_averages(short_names...; output_writer)
-
-
-Return a list of `ScheduledDiagnostics` that compute the hourly average for the given variables.
-"""
-hourly_averages(short_names...; output_writer) = common_diagnostics(
-    60 * 60,
-    (+),
-    output_writer,
-    short_names...;
-    pre_output_hook! = average_pre_output_hook!,
-)
-
-"""
-    hourly_average(short_names...; output_writer)
-
-
-Return a `ScheduledDiagnostics` that computes the hourly average for the given variable.
-"""
-hourly_average(short_names; output_writer) =
-    hourly_averages(short_names; output_writer)[1]
+function frequency_averages(t_end::Real)
+    if t_end > 365 * 86400
+        return monthly_averages
+    elseif t_end > 120 * 86400
+        return tendaily_averages
+    elseif t_end > 86400
```

Comment:
```suggestion
    elseif t_end >= 86400
```
(Just me being picky and following the logic [here](https://github.com/CliMA/ClimaAtmos.jl/pull/2605#issuecomment-1915792577) lol)

Reply:
Actually I don't mind too much. Just the first one `if t_end > 365 * 86400` I think it would be good to have 360 or whatever divisible by 30 .

Reply:
> (Just me being picky and following the logic https://github.com/CliMA/ClimaAtmos.jl/pull/2605#issuecomment-1915792577 lol)

I read your comment and ended up implementing something compeltely different :/